File: frontend/src/services/finance-dashboard.service.ts
/**
 * Finance Dashboard API Service
 * Story 8.6: Finance Dashboard
 *
 * Provides API client methods for finance dashboard endpoints
 */

import { apiClient } from '@/lib/api';
import {
  FinanceDashboard,
  FinanceKpi,
  IncomeExpenseChartData,
  ExpenseCategoryData,
  OutstandingReceivables,
  RecentTransaction,
  PdcStatusSummary,
} from '@/types/finance-dashboard';

const BASE_URL = '/api/v1/dashboard/finance';

/**
 * Finance Dashboard Service
 * API client methods for all finance dashboard endpoints
 */
export const financeDashboardService = {
  /**
   * Get complete finance dashboard data (AC-10)
   *
   * @param propertyId - Optional property filter
   * @returns Complete finance dashboard DTO
   */
  async getFinanceDashboard(propertyId?: string): Promise<FinanceDashboard> {
    const params = propertyId ? { propertyId } : {};
    const response = await apiClient.get<FinanceDashboard>(BASE_URL, { params });
    return response.data;
  },

  /**
   * Get finance KPIs (AC-1 to AC-4)
   * This is a convenience method - KPIs are also included in getFinanceDashboard
   *
   * @param propertyId - Optional property filter
   * @returns Finance KPI DTO
   */
  async getFinanceKpis(propertyId?: string): Promise<FinanceKpi> {
    const dashboard = await this.getFinanceDashboard(propertyId);
    return dashboard.kpis;
  },

  /**
   * Get income vs expense chart data (AC-5, AC-11)
   *
   * @param propertyId - Optional property filter
   * @returns List of monthly income vs expense data
   */
  async getIncomeVsExpense(propertyId?: string): Promise<IncomeExpenseChartData[]> {
    const params = propertyId ? { propertyId } : {};
    const response = await apiClient.get<IncomeExpenseChartData[]>(
      `${BASE_URL}/income-vs-expense`,
      { params }
    );
    return response.data;
  },

  /**
   * Get expense categories breakdown (AC-6, AC-12)
   *
   * @param propertyId - Optional property filter
   * @returns List of expense categories with amounts
   */
  async getExpenseCategories(propertyId?: string): Promise<ExpenseCategoryData[]> {
    const params = propertyId ? { propertyId } : {};
    const response = await apiClient.get<ExpenseCategoryData[]>(
      `${BASE_URL}/expense-categories`,
      { params }
    );
    return response.data;
  },

  /**
   * Get outstanding receivables summary (AC-7, AC-13)
   *
   * @param propertyId - Optional property filter
   * @returns Outstanding receivables with aging breakdown
   */
  async getOutstandingReceivables(propertyId?: string): Promise<OutstandingReceivables> {
    const params = propertyId ? { propertyId } : {};
    const response = await apiClient.get<OutstandingReceivables>(
      `${BASE_URL}/outstanding-receivables`,
      { params }
    );
    return response.data;
  },

  /**
   * Get recent high-value transactions (AC-8, AC-14)
   *
   * @param threshold - Minimum amount threshold (default 10000 AED)
   * @param propertyId - Optional property filter
   * @returns List of recent high-value transactions
   */
  async getRecentTransactions(
    threshold: number = 10000,
    propertyId?: string
  ): Promise<RecentTransaction[]> {
    const params: Record<string, string | number> = { threshold };
    if (propertyId) {
      params.propertyId = propertyId;
    }
    const response = await apiClient.get<RecentTransaction[]>(
      `${BASE_URL}/recent-transactions`,
      { params }
    );
    return response.data;
  },

  /**
   * Get PDC status summary (AC-9, AC-15)
   *
   * @param propertyId - Optional property filter
   * @returns PDC status summary DTO
   */
  async getPdcStatus(propertyId?: string): Promise<PdcStatusSummary> {
    const params = propertyId ? { propertyId } : {};
    const response = await apiClient.get<PdcStatusSummary>(
      `${BASE_URL}/pdc-status`,
      { params }
    );
    return response.data;
  },
};

export default financeDashboardService;

File: backend/src/main/java/com/ultrabms/service/FinanceDashboardService.java
package com.ultrabms.service;

import com.ultrabms.dto.dashboard.finance.*;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

/**
 * Service interface for Finance Dashboard operations.
 * Provides methods for financial KPIs, charts, and summaries.
 *
 * Story 8.6: Finance Dashboard
 */
public interface FinanceDashboardService {

    /**
     * Get complete finance dashboard data (AC-10)
     *
     * @param propertyId Optional property filter
     * @return Complete finance dashboard DTO
     */
    FinanceDashboardDto getFinanceDashboard(UUID propertyId);

    /**
     * Get finance KPIs (AC-1, AC-2, AC-3, AC-4)
     *
     * @param propertyId Optional property filter
     * @return Finance KPI DTO with YTD metrics
     */
    FinanceKpiDto getFinanceKpis(UUID propertyId);

    /**
     * Get income vs expense chart data (AC-5, AC-11)
     *
     * @param propertyId Optional property filter
     * @return List of monthly income vs expense data
     */
    List<IncomeExpenseChartDto> getIncomeVsExpense(UUID propertyId);

    /**
     * Get expense categories breakdown (AC-6, AC-12)
     *
     * @param propertyId Optional property filter
     * @return List of expense categories with amounts
     */
    List<ExpenseCategoryDto> getExpenseCategories(UUID propertyId);

    /**
     * Get outstanding receivables summary (AC-7, AC-13)
     *
     * @param propertyId Optional property filter
     * @return Outstanding receivables with aging breakdown
     */
    OutstandingReceivablesDto getOutstandingReceivables(UUID propertyId);

    /**
     * Get recent high-value transactions (AC-8, AC-14)
     *
     * @param threshold Minimum amount threshold (default 10000 AED)
     * @param propertyId Optional property filter
     * @return List of recent high-value transactions
     */
    List<RecentTransactionDto> getRecentTransactions(BigDecimal threshold, UUID propertyId);

    /**
     * Get PDC status summary (AC-9, AC-15)
     *
     * @param propertyId Optional property filter
     * @return PDC status summary DTO
     */
    PdcStatusSummaryDto getPdcStatus(UUID propertyId);
}
